tests/Dockerfile

```diff
@@ -0,0 +1,24 @@
+FROM ubuntu:24.04
+
+# Install Rust and required packages
+RUN apt-get update && DEBIAN_FRONTEND=noninteractive apt-get install -y \
+    curl \
+    build-essential \
+    pkgconf \
+    libssl-dev \
+    lsb-release \
+    kmod \
+    python3 \
+    python3-pip \
+    && rm -rf /var/lib/apt/lists/*
+
+# Install Rust
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
```

Comment:
It might be better to avoid the pipe to sh install method since we can also just `apt install rustup`. You can also omit the next line to set the path and do `rustup default stable` instead. Both methods seem to result in getting 1.82.0 for me.

Reply:
This method is recommended by Rust Team: https://www.rust-lang.org/tools/install
Do you think we should use apt instead?

Reply:
It's an 800+ line shell script that you pull from an external source, vs a package that is reviewed and signed off by a small group of individuals that have upload access to the Ubuntu archives.

It's "recommended" because it's the easiest thing that they can do to reach the widest audience regardless of what OS/distro they use, but in general, it's probably better to find another option if there is one. The deb package here gives you the same version of rust in the end too.

Reply:
Alternatively since this is just for running the tests themselves, there is an [official rust image](https://hub.docker.com/_/rust) that you could use as a base and then all you should need to do is copy over your tests set your entry point?

Reply:
I've tried it but it's debian-based and we need ubuntu-based environment

Reply:
I've updated the Dockerfile to use `apt install rustup`